Front-end theme: header

jefferya opened the pull request:
refs #372 

Copy over the header theme from Avalon 5 as per #372; use image instead of text for ERA-A+V header log

- library and ERA links: left-hand side of header
- add image as the ERA-AV logo in the middle of the header
- update styling of the login
- retain styling for small device screens via Bootstrap & update nav bar logo
- update search label
- update tests to reference ERA-AV

murny commented on app/helpers/era_av_helper.rb:
```diff
@@ -0,0 +1,29 @@
+    link_to("", "https://library.ualberta.ca", :class => 'logo')
+  end
+
+  def era_av_logo_link
```
Something I pushed back on chris a long time ago. 

But why not always have the logo as a link to the root_path? Why complicate this and change it's behaviour depending if your on the homepage or elsewhere on the site. It's extra logic that goes against what every site on the planet does (google, facebook, github, you name it, etc. all have their logos link back to the homepage, even if they are already on the homepage).

Then this helper isn't needed either and this whole file can get deleted.

murny commented on app/assets/stylesheets/era_custom.scss:
```diff
@@ -16,3 +22,147 @@ div.terms-of-use-custom {
+  font-size: 90%;
+}
+
+#masthead {
```
Again, i'd move this into a `header.scss` file.

Also ID's should almost never be used in CSS or JS. You already have a `custom-header` class to work with from the html, Could change all these ID references to use the `custom-header` class and nothing would change.

murny replied:
Also I'd take a look at how bootstrap does it's headers/navbar. Instead of writing all this custom CSS to do something similar.

murny commented on app/views/modules/_header.html.erb:
```diff
@@ -13,26 +13,28 @@ Unless required by applicable law or agreed to in writing, software distributed
+
+      <%# By default shift log in / log out to the header %>
+      <div class="col-sm-3">
+        <%= render partial: "modules/user_util_dropdowns"%>
```
The partial isn't that big, and this file itself isn't that big. Could just plop the code right here and get rid of the extra partial

murny commented on app/views/modules/_user_util_dropdowns.html.erb:
```diff
@@ -0,0 +1,12 @@
+    <div class="btn-group user_utility_links">
+      <a class="btn btn-default dropdown-toggle" data-toggle="dropdown" href="#"><span class="hidden-xs">&nbsp;<%= user_key %>&nbsp;</span><span class="sr-only">select for additional menu options</span><span class="caret"></span></a>
+      <ul class="dropdown-menu dropdown-menu-right" role="menu">
+        <li><%= link_to raw('<span class="glyphicon glyphicon-log-out"></span> log out'), destroy_user_session_path %></li>
```
instead of rendering using `raw` and a string, make your `link_to` a block and you can pass in the markup you want instead.

```
<%= link_to  destroy_user_session_path do %>
  <span class="glyphicon glyphicon-log-out"></span>
  Log out
<% end %>
```

(I capitalised `log out` to `Log out` as well, to be consistent with `Login`)

murny commented on app/views/layouts/avalon.html.erb:
```diff
@@ -19,13 +19,20 @@ Unless required by applicable law or agreed to in writing, software distributed
 
    <%= csrf_meta_tags %>
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
     <%= favicon_link_tag %>
```
Should maybe try to consolidate the two `favicon.ico` we have (/app/assets/ and /public folder into one?). 

I believe the default location the `favicon_link_tag` uses it the `/app/assets/` one, so you could just put in the actual markup to use the /public one, like you did below on line 31-35 (<link rel="icon" type="image/png" href="/favicon-16x16.png" sizes="16x16" />)

jefferya commented:
@murny refactored the following
- CSS: removed ID selectors; refactored to use more Bootstrap features
- refactored partials and helpers: small code chunks not being reused included in view
- Consolidated favicon into one location
- link_to helper with block instead of using `raw`

@murny any additional suggestions regarding Bootstrap & CSS - I'm not sure if there is more that can be done within Bootsrap 3.1.3.

murny commented on app/assets/stylesheets/uofa_header.scss:
```diff
@@ -0,0 +1,74 @@
+  background-color: $header-background;
+}
+
+// UAL logo change colour on hover by changing the background position
```
Can just leave this then! Probably not a good way to refactor this. 

(And honestly, I'd probably just push to simply this. Why have a janky switching of logos when you hover over it? Especially when you can't hover on mobile! Should just use one logo all the time (like most sites around the world) and then none of this fancy CSS is needed)
